src/GoodsList.tsx

```
@@ -1,16 +1,46 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Good } from './types/Good';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
 type Props = {
-  goods: Good[];
+  sortBy: string;
 };
 
-export const GoodsList: React.FC<Props> = ({ goods }) => (
-  <ul>
-    {goods.map(good => (
-      <li key={good.id} data-cy="good">
-        {good.name}
-      </li>
-    ))}
-  </ul>
-);
+export const GoodsList: React.FC<Props> = ({ sortBy }) => {
+  const [goods, setGoods] = useState<Good[]>([]);
+
+  useEffect(() => {
+    if (!sortBy) {
+      return;
+    }
+
+    const fetchGoods = async () => {
+      switch (sortBy) {
+        case 'all':
+          setGoods(await getAll());
+          break;
+        case 'red':
+          setGoods(await getRedGoods());
+          break;
+        case 'first-five':
+          setGoods(await get5First());
```

This violates checklist item #3: "3. [API] - When you make a request to the server, don't forget to handle possible errors". The `fetchGoods` function calls `await getAll()`, `await getRedGoods()`, and `await get5First()` without try/catch, so any rejection will be uncaught here and the UI will not show a clear error state. Wrap these calls with try/catch and set an error state (or ensure the API helpers catch and return errors in a predictable form) so the UI can surface failures instead of failing silently.